x-pack/plugins/siem/public/components/page/add_to_kql/index.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiIcon, EuiPanel, EuiToolTip } from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import React from 'react';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+
+import { HostsFilter } from '../../../containers/hosts';
+import { NetworkFilter } from '../../../containers/network';
+import { WithSource } from '../../../containers/source';
+import { IndexType } from '../../../graphql/types';
+import { hostsModel, KueryFilterQuery, networkModel } from '../../../store';
+import { WithHoverActions } from '../../with_hover_actions';
+
+import * as i18n from './translations';
+
+interface Props {
+  applyFilterQueryFromKueryExpression: (expression: string) => void;
+  children: JSX.Element;
+  expression: string;
+  filterQueryDraft: KueryFilterQuery;
+}
+
+class AddToKqlComponent extends React.PureComponent<Props> {
+  public render() {
+    const { children } = this.props;
+    return (
+      <WithHoverActions
+        hoverContent={
+          <HoverActionsContainer data-test-subj="hover-actions-container">
+            <EuiToolTip content={i18n.FILTER_TO_KQL}>
+              <EuiIcon type="filter" onClick={this.addToKql} />
+            </EuiToolTip>
+          </HoverActionsContainer>
+        }
+        render={() => children}
+      />
+    );
+  }
+
+  private addToKql = () => {
+    const { expression, filterQueryDraft, applyFilterQueryFromKueryExpression } = this.props;
+    applyFilterQueryFromKueryExpression(
+      filterQueryDraft && !isEmpty(filterQueryDraft.expression)
+        ? `${filterQueryDraft.expression} and ${expression}`
+        : expression
+    );
+  };
+}
+
+const HoverActionsContainer = styled(EuiPanel)`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 25px;
+  justify-content: center;
+  left: 5px;
+  position: absolute;
+  top: -10px;
+  width: 30px;
+  cursor: pointer;
+`;
```

Comment:
Just a note, no changes requested. I've been avoiding using `flex` directly in favor of the `EuiFlexItem` for when/if we have to port over to `IE-11` I would imagine using the Eui components would be easier.

It might not be possible to use it in this case, just putting out in case it is possible it would probably be better to use those components?